backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/disk/image/ConvertDiskCommand.java

```diff
@@ -0,0 +1,366 @@
+                getDiskImage().getStorageIds().get(0),
+                getDiskImage().getId(),
+                getParameters().getNewVolGuid());
+
+        DiskImage newImage = DiskImage.copyOf(getDiskImage());
+
+        newImage.setImageId(getParameters().getNewVolGuid());
+        if (getParameters().getVolumeFormat() != null) {
+            if (info.getVolumeFormat() != getParameters().getVolumeFormat()) {
+                log.error("Requested format '{}' doesn't match format on storage '{}'",
+                        getParameters().getVolumeFormat(), info.getVolumeFormat());
+                setCommandStatus(CommandStatus.FAILED);
+                return;
+            }
+
+            newImage.setVolumeFormat(getParameters().getVolumeFormat());
+        }
+
+        if (getParameters().getPreallocation() != null) {
+            if (info.getVolumeFormat() != getParameters().getVolumeFormat()) {
+                log.error("Requested allocation policy '{}' doesn't match allocation policy on storage '{}'",
+                        getParameters().getPreallocation(), info.getVolumeType());
+                setCommandStatus(CommandStatus.FAILED);
+                return;
+            }
+
+            newImage.setVolumeType(getParameters().getPreallocation());
+        }
+
+        getDiskImage().setActive(false);
+        newImage.getImage().setSequenceNumber(imageDao.getMaxSequenceNumber(getDiskImage().getId()) + 1);
+        TransactionSupport.executeInNewTransaction(() -> {
+            addDiskImageToDb(newImage, getCompensationContext(), true);
+            imageDao.update(getDiskImage().getImage());
+
+            return null;
+        });
+    }
+
+    private void updatePhase(ConvertDiskCommandParameters.ConvertDiskPhase phase) {
+        getParameters().setConvertDiskPhase(phase);
+        persistCommandIfNeeded();
+    }
+
+    private VolumeType getVolumeType(DiskImage diskImage) {
+        return getParameters().getPreallocation() == null ? diskImage.getVolumeType() : getParameters().getPreallocation();
+    }
+
+    private VolumeFormat getVolumeFormat(DiskImage diskImage) {
+        return getParameters().getVolumeFormat() == null ? diskImage.getVolumeFormat() : getParameters().getVolumeFormat();
+    }
+
+    private VdsmImageLocationInfo getLocationInfo() {
+        return (VdsmImageLocationInfo) getParameters().getLocationInfo();
+    }
+
+    @Override
+    protected void setActionMessageParameters() {
+        super.setActionMessageParameters();
+        addValidationMessage(EngineMessage.VAR__ACTION__CONVERT_DISK);
+        addValidationMessage(EngineMessage.VAR__TYPE__DISK);
+    }
+
+    @Override
+    protected Map<String, Pair<String, String>> getExclusiveLocks() {
+        return Collections.singletonMap(getDiskImage().getId().toString(),
+                LockMessagesMatchUtil.makeLockingPair(LockingGroup.DISK, EngineMessage.ACTION_TYPE_FAILED_DISK_IS_LOCKED));
```

Comment:
it would be better to set a different lock message which indicates that the disk is being converted

Reply:
ack, created a new message